capabilities/README.md

```diff
@@ -1,91 +1,190 @@
 
-libcap
+Capabilities apply to both files and threads. File capabilities allow users to execute programs with higher privileges. This is similar to the way the `setuid` bit works. Thread capabilities keep track of the current state of capabilities in running programs.
 
-* capsh - lets you modify capabilities within its own process in arbitrary ways and then execute a shell
-* setcap - set capability bits on a file
-  * `setcap cap_sys_admin+pie ls`
-  * `setcap -r ls`
-* getcap - get the capability bits from a file
-  * `getcap -r .`
+The Linux kernel lets you set capability *bounding sets* that impose limits on the capabilities that a file/thread can gain.
 
-libcap-ng
+Docker imposes certain limitations that make working with capabilities much simpler. For example, file capabilities are stored within a file's extended attributes, and extended attributes are stripped out when Docker images are built. This means you will not normally have to concern yourself too much with file capabilities in containers.
 
-* pscap - list the capabilities of running processes
-* filecap - list the capabilities of files
-* captest - check what capabilities are available to the captest process
+> It is of course possible to get files with capabilities into running containers via volumes, bind mounts, or adding files during run time. However, this is not recommended.
 
-#### More libcap/libcap-ng usage examples
+**NOTE TO LAB OWNER: I pretty much took the above recommendation form the original lab text. However, I'm not sure it sounds right as it could sound like we're not recommending volumes etc. You might wanna make some changes to the way I've written that.**
 
-**Printing out all capabilities**
+In an environment without file based capabilities, it's not possible for applications to escalate their privileges beyond the *bounding set* (a set beyond which capabilities cannot grow). Docker sets the *bounding set* before starting a container. You can use Docker commands to add or remove capabilities to or from the *bounding set*.
 
-In alpine you need to install the libcap package to use capsh:
-```
-$ docker run --rm -it alpine sh -c 'apk add -U libcap; capsh --print'
-Current: = cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_bind_service,cap_net_raw,cap_sys_chroot,cap_mknod,cap_audit_write,cap_setfcap+eip
-Bounding set =cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_bind_service,cap_net_raw,cap_sys_chroot,cap_mknod,cap_audit_write,cap_setfcap
-Securebits: 00/0x0/1'b0
- secure-noroot: no (unlocked)
- secure-no-suid-fixup: no (unlocked)
- secure-keep-caps: no (unlocked)
-uid=0(root)
-gid=0(root)
-groups=0(root),1(bin),2(daemon),3(sys),4(adm),6(disk),10(wheel),11(floppy),20(dialout),26(tape),27(video)
-```
+**NOTE TO LAB OWNER: I added a quick description of bounding set but I'm no kernel hacker so might be off the mark....**
```

Comment:
no problem, it sounds pretty good!  I'll add Docker's bounding set as a link to help illustrate 👍